gamepad/gamepad/publisher.py

```diff
@@ -0,0 +1,206 @@
+                        self.home = ('Select:0 Start:0 Home:' + str(event.state))
+                        msg_misc = String()
+                        msg_misc.data = str(self.home)
+                        self.miscellaneous.publish(msg_misc)
+                        self.get_logger().info(msg_misc.data)
+                    elif event.code == 'BTN_EAST' and event.state == 1 or event.code == 'BTN_EAST' and event.state == 0:
+                        os.system('clear')
+                        self.xbox_controller()
+        elif self.val == '6':
+            print('Joystick \n(i.e. to go back to the menu press the \"Xbox Home Button\")')
+            while 1:
+                events = get_gamepad()
+                for event in events:
+                    self.joystick_value = str(event.state)
+                    if event.code == "ABS_RY" and event.state < 0:
+                        self.location = ('Right joy stick left or up ' + self.joystick_value)
+                        msg_joy = String()
+                        msg_joy.data = str(self.location)
+                        self.joystick.publish(msg_joy)
+                        self.get_logger().info(msg_joy.data)
+                        #print("Right joy stick right or up " + joystick_value, end='\r')
+                    elif event.code == "ABS_RY" and event.state > 0:
+                        self.location = ('Right joy stick down or right ' + self.joystick_value)
+                        msg_joy = String()
+                        msg_joy.data = str(self.location)
+                        self.joystick.publish(msg_joy)
+                        self.get_logger().info(msg_joy.data)
+                        #print("Right joy stick down or left " + joystick_value, end='\r')
+                    elif event.code ==  "ABS_Y" and event.state > 0:
+                        self.location = ('Left joy stick down or right ' + self.joystick_value)
+                        msg_joy = String()
+                        msg_joy.data = str(self.location)
+                        self.joystick.publish(msg_joy)
+                        self.get_logger().info(msg_joy.data)
+                        #print("Left joy stick down or right " + joystick_value, end='\r')
+                    elif event.code == "ABS_Y" and event.state < 0:
+                        self.location = ('Left joy stick up or left ' + self.joystick_value)
+                        msg_joy = String()
+                        msg_joy.data = str(self.location)
+                        self.joystick.publish(msg_joy)
+                        self.get_logger().info(msg_joy.data)
+                        #print("Left joy stick up or left " + joystick_value, end='\r')
+                    elif event.code == "BTN_MODE" and event.state == 1:
+                        os.system('clear')
+                        self.xbox_controller()
+                
```

Comment:
It might be a good idea to include an `else`, just in case a different input is received. This way you can let the user know that the given choice is invalid